x-pack/plugins/security_solution/common/api/timeline/model/components.gen.ts

```diff
@@ -50,46 +57,33 @@ export const ColumnHeaderResult = z.object({
 
 export type QueryMatchResult = z.infer<typeof QueryMatchResult>;
 export const QueryMatchResult = z.object({
-  field: z.string().optional(),
-  displayField: z.string().optional(),
-  value: z.string().optional(),
-  displayValue: z.string().optional(),
-  operator: z.string().optional(),
+  field: z.string().nullable().optional(),
+  displayField: z.string().nullable().optional(),
+  value: z.string().nullable().optional(),
+  displayValue: z.string().nullable().optional(),
+  operator: z.string().nullable().optional(),
 });
 
-export interface DataProviderResult {
-  id?: string;
-  name?: string;
-  enabled?: boolean;
-  excluded?: boolean;
-  kqlQuery?: string;
-  queryMatch?: QueryMatchResult;
-  and?: DataProviderResult[];
-  type?: DataProviderType;
-}
-export interface DataProviderResultInput {
-  id?: string;
-  name?: string;
-  enabled?: boolean;
-  excluded?: boolean;
-  kqlQuery?: string;
-  queryMatch?: QueryMatchResult;
-  and?: DataProviderResultInput[];
-  type?: DataProviderType;
-}
-export const DataProviderResult: z.ZodType<
-  DataProviderResult,
-  ZodTypeDef,
-  DataProviderResultInput
-> = z.object({
-  id: z.string().optional(),
-  name: z.string().optional(),
-  enabled: z.boolean().optional(),
-  excluded: z.boolean().optional(),
-  kqlQuery: z.string().optional(),
+export type DataProviderQueryMatch = z.infer<typeof DataProviderQueryMatch>;
+export const DataProviderQueryMatch = z.object({
+  enabled: z.boolean().nullable().optional(),
+  excluded: z.boolean().nullable().optional(),
+  id: z.string().nullable().optional(),
+  kqlQuery: z.string().nullable().optional(),
+  name: z.string().nullable().optional(),
   queryMatch: QueryMatchResult.optional(),
-  and: z.array(z.lazy(() => DataProviderResult)).optional(),
-  type: DataProviderType.optional(),
+});
+
+export type DataProviderResult = z.infer<typeof DataProviderResult>;
```

Comment:
Ah, nice catch here. I'm pretty sure I based the original version of this type [here](https://github.com/elastic/kibana/blob/ecc2e4e2a409ab18b3017412a9b0af229b83b884/x-pack/plugins/security_solution/common/api/timeline/model/api.ts#L601), but that is slightly different from the validation schema. 👌🏾